src/useWorker.ts

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable no-console */
+import React from 'react'
+import createWorkerBlobUrl from './lib/createWorkerBlobUrl'
+import WORKER_STATUS from './lib/status'
+
+type Options = {
+  timeout?: number;
+  dependencies: string[];
+}
+
+const PROMISE_RESOLVE = 'resolve'
+const PROMISE_REJECT = 'reject'
+const DEFAULT_OPTIONS: Options = {
+  timeout: undefined,
+  dependencies: [],
+}
+
+/**
+ *
+ * @param {Function} fn the function to run with web worker
+ * @param {Object} options useWorker option params
+ */
+export const useWorker = <T extends (...fnArgs: any[]) => any>(fn: T, options: Options = DEFAULT_OPTIONS) => {
+  const [workerStatus, setWorkerStatus] = React.useState<WORKER_STATUS>(WORKER_STATUS.PENDING)
+  const worker = React.useRef<Worker & { _url?: string }>()
+  const promise = React.useRef<{ [PROMISE_REJECT]?: (result: ReturnType<T> | ErrorEvent) => void;[PROMISE_RESOLVE]?: (result: ReturnType<T>) => void }>({})
+  const timeoutId = React.useRef<number>()
+
+  const killWorker = (status = WORKER_STATUS.PENDING) => {
+    if (worker.current && worker.current._url) {
```

Comment:
```suggestion
    if (worker.current?._url) {
```

Reply:
the build tool doesn't support optional chaining with TS :(

Reply:
We can install `@babel/plugin-proposal-optional-chaining` and adding the plugin into `.babelrc`, It should work

**.babelrc**
```json
{
    "plugins": [
        [
            "@babel/plugin-proposal-optional-chaining",
            {
                "loose": false
            }
        ]
    ]
}
```

I can take care of it, once we merged this PR 👍

cc: @IljaDaderko @z4o4z